src/implementations/svd.jl

```diff
@@ -365,54 +335,65 @@ function svd_full!(A::AbstractMatrix, USVᴴ, alg::GPU_SVDAlgorithm)
         one!(Vᴴ)
         return USVᴴ
     end
+
+    do_gauge_fix = get(alg.kwargs, :gaugefix, default_gaugefix())::Bool
+    lapack_kwargs = Base.structdiff(alg.kwargs, NamedTuple{(:gaugefix,)})
+
     if alg isa GPU_QRIteration
-        isempty(alg.kwargs) ||
-            @warn "GPU_QRIteration does not accept any keyword arguments"
+        isempty(lapack_kwargs) || @warn "invalid keyword arguments for GPU_QRIteration"
         _gpu_gesvd_maybe_transpose!(A, view(S, 1:minmn, 1), U, Vᴴ)
     elseif alg isa GPU_SVDPolar
-        _gpu_Xgesvdp!(A, view(S, 1:minmn, 1), U, Vᴴ; alg.kwargs...)
+        _gpu_Xgesvdp!(A, view(S, 1:minmn, 1), U, Vᴴ; lapack_kwargs...)
     elseif alg isa GPU_Jacobi
-        _gpu_gesvdj!(A, view(S, 1:minmn, 1), U, Vᴴ; alg.kwargs...)
+        _gpu_gesvdj!(A, view(S, 1:minmn, 1), U, Vᴴ; lapack_kwargs...)
     else
         throw(ArgumentError("Unsupported SVD algorithm"))
     end
     diagview(S) .= view(S, 1:minmn, 1)
     view(S, 2:minmn, 1) .= zero(eltype(S))
-    # TODO: make this controllable using a `gaugefix` keyword argument
-    gaugefix!(svd_full!, U, S, Vᴴ, m, n)
+
+    do_gauge_fix && gaugefix!(svd_full!, U, Vᴴ)
+
     return USVᴴ
 end
 
 function svd_trunc!(A::AbstractMatrix, USVᴴ, alg::TruncatedAlgorithm{<:GPU_Randomized})
     check_input(svd_trunc!, A, USVᴴ, alg.alg)
     U, S, Vᴴ = USVᴴ
     _gpu_Xgesvdr!(A, S.diag, U, Vᴴ; alg.alg.kwargs...)
-    # TODO: make this controllable using a `gaugefix` keyword argument
-    gaugefix!(svd_trunc!, U, S, Vᴴ, size(A)...)
+
     # TODO: make sure that truncation is based on maxrank, otherwise this might be wrong
-    USVᴴtrunc, ind = truncate(svd_trunc!, (U, S, Vᴴ), alg.trunc)
-    Strunc = diagview(USVᴴtrunc[2])
+    (Utr, Str, Vᴴtr), _ = truncate(svd_trunc!, (U, S, Vᴴ), alg.trunc)
+
     # normal `truncation_error!` does not work here since `S` is not the full singular value spectrum
-    ϵ = sqrt(norm(A)^2 - norm(Strunc)^2) # is there a more accurate way to do this?
-    return USVᴴtrunc..., ϵ
+    ϵ = sqrt(norm(A)^2 - norm(diagview(Str))^2) # is there a more accurate way to do this?
+
+    do_gauge_fix = get(alg.alg.kwargs, :gaugefix, default_gaugefix())::Bool
+    do_gauge_fix && gaugefix!(svd_trunc!, Utr, Vᴴtr)
+
+    return Utr, Str, Vᴴtr, ϵ
 end
 
 function svd_compact!(A::AbstractMatrix, USVᴴ, alg::GPU_SVDAlgorithm)
     check_input(svd_compact!, A, USVᴴ, alg)
     U, S, Vᴴ = USVᴴ
+
+    do_gauge_fix = get(alg.kwargs, :gaugefix, default_gaugefix())::Bool
+    lapack_kwargs = Base.structdiff(alg.kwargs, NamedTuple{(:gaugefix,)})
```

Comment:
I don't remember if I made this comment earlier, but would it make sense to call all the `lapack_kwargs` something like `alg_kwargs`, `computational_kwargs` or `driver_kwargs`, since this is not even LAPACK anymore at this point?